extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -132,13 +125,13 @@ static io.deephaven.qst.type.Type<?> convertPrimitiveType(final Type icebergType
     }
 
     @SuppressWarnings("unused")
-    public List<TableIdentifier> listTables(final Namespace namespace) {
+    public List<TableIdentifier> listTables(@NotNull final Namespace namespace) {
         // TODO: have this return a Deephaven Table of table identifiers
         return catalog.listTables(namespace);
     }
 
     public List<Long> listTableSnapshots(@NotNull final TableIdentifier tableIdentifier) {
-        final ArrayList<Long> snapshotIds = new ArrayList<>();
+        final List<Long> snapshotIds = new ArrayList<>();
```

Comment:
I think a few things:
1. It's better to mirror the Iceberg name/interface. That is, we should offer `Iterable<Snapshot> snapshots()` or `List<Snapshot> snapshots()`.
2. We should have a method to get a Deephaven table of snapshots, expanding to show sequence number, ID, timestamp, parent, etc.